File: Python/CustomSolution.py
def primality_test(num):
    if num <= 1:
        return False
    if num%2 == 0:
        return num==2
    i = 3
    while(i*i<=num):
        if num%i == 0:
            return False
        i+=2
    return True

test = int(input("Enter a number to check if it's prime or composite: "))
print("The number is "+("prime" if primality_test(test) else "not prime"))

File: Java/CustomSolution.java
import java.util.Scanner;

public class Main {
    public static boolean PrimalityTest(int num) {
        if (num <= 1) return false;
        if (num%2 == 0) return num==2;
        for (int i = 3; i*i<=num; i+=2)
            if (num%i==0)
                return false;
        return true;
    }

    public static void main(String[] args) {
        System.out.println("Enter a number to check if it's prime or composite");
        Scanner input = new Scanner(System.in);
        int num = input.nextInt();
        System.out.println("The number you entered is "+(PrimalityTest(num)?"prime":"not prime"));
    }
}
